### PR TITLE
fix(a11y): disabled button behavior and color contrast

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

## Issues and Context

There are two issues with the Button component, which were brought up in https://github.com/freeCodeCamp/freeCodeCamp/pull/53824#issuecomment-2018611111:
1. Button with `type="submit"` can still trigger form submission even if it's disabled
2. The text of disabled button doesn't have sufficient contrast

(1) is indeed a bug and the fix is straight forward, so I only want to discuss (2).

So if we want to reflect the disabled state of an element, we can use either the HTML `disabled` or `aria-disabled` attribute. 

With the `disabled` attribute, the element is still present in the DOM, but removed from the accessibility tree, meaning non-sighted users won't know the button is there nor its disabled state, and they also can't interact with it.

With the `aria-disabled` attribute, the users can use the keyboard to focus on the button, and can be informed about the element and its disabled state. This is the behavior I decided to go with, as the UX is much better.

However, we have been dimming the text and background color of the button is communicate the disabled state visually. This would be acceptable if the button isn't interactable, but since we want it to be interactable, we are failing the color contrast requirement.

## Approaches

To address the color contrast requirement, the most straightforward fix would be improving the contrast. We are currently using the `opacity` attribute to dim the colors, but in order to make the button look nice and accessible, we would need to play with the color combinations and not just cranking up the opacity. We also don't have colors that could be used for the disabled state of the `danger` and `info` variants, so I'd need some help here (from Ahmad) if we want to proceed with the color update.

Alternatively, I think we should consider removing the disabled state and keeping buttons enabled at all time. 
- If the user isn't allowed to perform an action, we can completely hide the button.
- If the user can only perform an action if they have fulfilled all the requirements, we can add a validation check to the button, and show a message telling the user what they have missed, or complete the action if the requirements are met.

That would help resolve the color contrast and interactivity issues, while also improving the user experience. (I think we've been assuming the users would understand why the button is disabled and how to enable it, but that might not be obvious for some).

## My Fix

I might have gone _a little_ overboard with the changes.

My fix is a mix of both approaches:
- I updated the CSS of the disabled state of the `primary` button to improve the contrast
- I set the `disabled` value of the `danger` and `info` to always `false`

Rationale: 
  - I checked /learn's `global.css` stylesheet and found that all buttons have `primary` styles when they are disabled (notice the `button` selector and the colors used in [the code](https://github.com/freeCodeCamp/freeCodeCamp/blob/086ff3633302c4702e620da99eb5454c027b1a91/client/src/components/layouts/global.css#L279-L282)), so I assume we implicitly don't support disabled state in `danger` and `info` variants (otherwise they look very different from their enabled state).
  - In /learn, there isn't any `danger` or `info` button that is used with the `disabled` prop (meaning `danger` and `info` buttons are enabled at all time).

The changes are to close down the feature that has no usages, and to move us closer to the second approach above (assuming everyone agrees with the approach).

## Screenshots

Basically instead of dimming the text and the background colors, I dim the border color to communicate the disabled state.

| | Before | After |
| --- | --- | --- |
| Light | Screenshot 2024-04-11 at 14 31 33 | Screenshot 2024-04-11 at 14 32 25 | 
| Dark | Screenshot 2024-04-11 at 14 31 41 | Screenshot 2024-04-11 at 14 32 17 |

### CODE REVIEW

**Review by ahmaxed — APPROVED** [review]

**Comment by ahmaxed** [comment]
We could go with the current colors and discuss adding more in the redesign.
